slick.grid.js

```diff
@@ -2916,13 +2916,8 @@ if (typeof Slick === "undefined") {
     function setColumns(columnDefinitions) {
       trigger(self.onBeforeSetColumns, { previousColumns: columns, newColumns: columnDefinitions, grid: self });
 
-      var _treeColumns = new Slick.TreeColumns(columnDefinitions);
-      if (_treeColumns.hasDepth()) {
-        treeColumns = _treeColumns;
-        columns = treeColumns.extractColumns();
-      } else {
-        columns = columnDefinitions;
-      }
+      treeColumns = new Slick.TreeColumns(columnDefinitions);
+      columns = treeColumns.extractColumns();
```

Comment:
we can hardly accept this PR, we actually were thinking to get rid of it, but you are trying to making it the only option. There's a long post about this in this issue #592, I still don't understand what this tree does exactly

Reply:
Whew! Just read through all of that for #592. So it looks like we never actually did get rid of TreeColumns? I do think there are much less complex ways of implementing column grouping than a tree structure, and given that the column array is a fundamental part of SlickGrid, it's best not messed with.